toolchain/lower/mangler.cpp

```diff
@@ -0,0 +1,91 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/lower/mangler.h"
+
+#include "toolchain/base/kind_switch.h"
+#include "toolchain/sem_ir/entry_point.h"
+
+namespace Carbon::Lower {
+
+auto Mangler::MangleInverseQualifiedNameScope(bool first_name_component,
```

Comment:
For my own understanding, why inverse instead of forward?

Reply:
Added a comment explaining this - I think someone, maybe @zygoloid, suggested that mangling in inverse order may reduce the incidence of names having a common prefix which might slow down uniqueness tests/comparisons (since the comparison would always have to walk the common prefixes before finding something unique).